website/pages/docs/deployment/docker-read-only.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: Docker - Read-only Containers
+description: Learn how to run containers with read-only file system.
+---
+
+# Docker - Read-only Containers
+
+By default, CloudQuery requires a read-write file system to create a socket file used by the plugins to communicate with the server. It is however not possible to create a read-write socket on a read-only file system. For this to work, a plugin needs to be started as a gRPC server locally.
+
+## Creating a container that works with a read-only file system
+
+To create a container that works with a read-only file system, you will need the following:
+
+- a CloudQuery config file that is used to install the required plugins
+- a CloudQuery config file that contains the specs needed for the sync
+- a start script used as an entry point for the container that will start the plugins and run the sync
+
+In the example below, we will use the [XKCD](https://github.com/hermanschaaf/cq-source-xkcd) source plugin and [PostgreSQL](https://hub.cloudquery.io/plugins/destination/cloudquery/postgresql/v6.1.3/docs) destination plugin.
+
+### Create a config file for the installation
+
+This config file should reference the plugins that you want to install with the actual source where they should be downloaded from:
+
+```yaml
+# install.yaml
+kind: source
+spec:
+  name: xkcd
+  path: hermanschaaf/xkcd
+  version: v2.0.0
+  destinations: ["postgresql"]
+  tables: ["*"]
+---
+kind: destination
+spec:
+  name: "postgresql"
+  registry: "github"
+  path: "cloudquery/postgresql"
+  version: "v6.1.3"
+  spec:
+    connection_string: "postgresql://cq:password123@host.docker.internal:5432/cq-test?sslmode=disable"
```

Comment:
Is the connection string actually needed here? I think you should be able to leave it empty when using with the `cloudquery plugin install` command. If not, we should file a bug for it

Reply:
no, I'll fix it